modes/preclinical-4d/src/index.tsx

```diff
@@ -62,36 +62,19 @@ function modeFactory({ modeConfiguration }) {
 
       // the primary button section is created in the workflow steps
       // specific to the step
-      customizationService.addModeCustomizations([
-        {
-          id: 'PanelSegmentation.tableMode',
-          mode: 'expanded',
+      customizationService.setCustomizations({
+        'panelSegmentation.tableMode': {
+          $set: 'expanded',
         },
-        {
-          id: 'PanelSegmentation.onSegmentationAdd',
-          onSegmentationAdd: () => {
+        'panelSegmentation.onSegmentationAdd': {
+          $set: () => {
```

Comment:
A simple assignment as a function should require the $set

Reply:
I know it involves more boilerplate code, but this approach consistently handles both simple cases and really complex deep nestings. I'd rather not have a long explanation of different setup methods and instead stick to one consistent way to `set` (replace). 

Basically the message to the user is if you want to customize use a $ operator.